.ci/test.sh

```diff
@@ -54,7 +54,7 @@ if [[ $TASK == "lint" ]]; then
     conda install -q -y -n $CONDA_ENV \
         pycodestyle \
         pydocstyle \
-        r-lintr
+        r-lintr==2.0.0
```

Comment:
@jameslamb 

Maybe `>=2.0.0` is better to free us from bumping the version after each new release?

> lintr released 2.0.0 to CRAN (and r-lintr to conda-forge).

```
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... failed with initial frozen solve. Retrying with flexible solve.
PackagesNotFoundError: The following packages are not available from current channels:
  - r-lintr==2.0.0
Current channels:
  - https://repo.anaconda.com/pkgs/main/linux-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/linux-64
  - https://repo.anaconda.com/pkgs/r/noarch
To search for alternate channels that may provide the conda package you're
looking for, navigate to
    https://anaconda.org
and use the search bar at the top of the page.
```

The you should either wait for release in default (`r`) channel, or install it from `conda-forge` channel:
```
conda install -q -y -n $CONDA_ENV -c conda-forge r-lintr
```

Reply:
@StrikerRUS ah good catch on the channels, my mistake

I think `==2.0.0` is actually better for this case. Since this is just a linter, I think it is desirable to have it keep performing the same and then at some point in the future (6 months or 1 year) update if they have a new release. Otherwise we risk some PR unrelated to the R code having its CI break because a new `lintr` release is found. So in other words...I think pegging to an exact version in this case is _less_ maintenance burden, not more.

Reply:
> Otherwise we risk some PR unrelated to the R code having its CI break because a new lintr release is found.

Can you please be more precise? In case of introducing of new linters which LightGBM code doesn't follow, then yes. But we have the exact list of checking linters, so I guess it's not possible. New versions will (potentially) bring bug fixes and speedups, which is good to have ASAP after release.

Reply:
I was able to fix this (I think!) by re-installing the `stringi` package. This was suggested [deep in the arch linux forum archive](https://bbs.archlinux.org/viewtopic.php?id=235861)

Reply:
Sorry, resolved this too fast. I didn't realize this was also the `==` vs . `>=` thread.

I am ok with using `>=` so that we get bugfixes and performance enhancements ASAP. I'll make that change.

Reply:
ok now I think this can actually be resolved. I'm using . `>=`